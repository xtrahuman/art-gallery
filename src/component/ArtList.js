import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { FaRegArrowAltCircleRight } from 'react-icons/fa';
import gallery from '../images/gallery.svg';
import Navbar from './Navbar';

const ArtList = () => {
  const art = useSelector((state) => state.artReducer);
  const [input, setInput] = useState('');
  const allArt = art.filter((art) => {
    let newArt;
    if (input === '') {
      newArt = art;
    }
    if (art.title.toLowerCase().includes(input.toLowerCase())) {
      newArt = art;
    }
    return newArt;
  });

  const Handlechange = (e) => {
    setInput(e.target.value);
  };
  return (
    <div className="checkArt">
      <Navbar name="Art List" />
      <div className="artContainer">
        <div className="ArtGallery">
          <div className="listImg">
            <img src={gallery} alt="art" />
          </div>
          <div className="artText">
            <h1>Art Gallery</h1>
            <p>
              {`${allArt.length} `}
              {`ArtWork${allArt.length <= 1 ? '' : 's'}`}
            </p>
          </div>
        </div>
        <div className="searchInput d-flex">
          <input placeholder="search art" onChange={Handlechange} />
        </div>
        <div className="d-flex justify-row">
          <div className="row">
            {allArt.map((art) => (
              <div key={art.id} className="artCont">
                <div className="artFlex">
                  <div className="img-f"><Link className="artTitle" to={`/details/${art.id}`}><img src={art.imgcheck === null ? gallery : art.imgt} alt="art" /></Link></div>
                  <div className="artLink">
                    <Link to={`/details/${art.id}`}><FaRegArrowAltCircleRight className="arrowCircle" /></Link>
                  </div>
                </div>
                <div>
                  <Link className="artTitle" to={`/details/${art.id}`}>
                    <p>{art.title}</p>
                    <p>{art.displayDate}</p>
                  </Link>
                </div>
              </div>
            ))}

          </div>
        </div>
      </div>
    </div>
  );
};
export default ArtList;
