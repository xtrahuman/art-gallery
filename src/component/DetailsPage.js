import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { eachArt } from '../redux/DetailsPage/DetailsPage';
import Navbar from './Navbar';

const DetailsPage = () => {
  const { id } = useParams();
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(eachArt(id));
  }, []);

  const art = useSelector((state) => state.detailReducer);
  return (
    <div className="checkDetails">
      <Navbar name="Art Details" />
      {art.map((art) => (
        <div key={art.id}>
          <div className="ArtGallery detailGallery">
            <div className="detailImg">
              <img src={art.imgt} alt="art" />
            </div>
            <div className="artText detailText">
              <h2>{art.title}</h2>
            </div>
          </div>
          <div>
            <p className="p-tile">{art.title}</p>
            <div className="d-flex detail-bar">
              <p>Display date</p>
              <p className="p-det">{art.displayDate}</p>
            </div>
            <div className="d-flex detail-bar">
              <p>Artist Profile</p>
              <p className="p-det">{art.artist}</p>
            </div>
            <div className="d-flex detail-bar">
              <p>Country of Origin</p>
              <p className="p-det">{art.country}</p>
            </div>
            <div className="d-flex detail-bar">
              <p>Credit</p>
              <p className="p-det">{art.credit}</p>
            </div>
          </div>
        </div>
      ))}
    </div>
  );
};

export default DetailsPage;
