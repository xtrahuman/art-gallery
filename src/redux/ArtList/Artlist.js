const GET_ART = 'art-gallery/Artlist/GET_ART';
const url = 'https://api.artic.edu/api/v1/artworks?page=1';
const intialState = [];
export const FetchArt = async () => {
  const request = await fetch(url);
  const artWork = await request.json();
  return artWork;
};

export const getArt = () => async (dispatch) => {
  const ArtDetail = await FetchArt();
  const collection = ArtDetail.data.map((data) => {
    const { title, id } = data;
    return {
      title,
      id,
      displayDate: data.date_display,
      imgcheck: data.image_id,
      imgt: `https://www.artic.edu/iiif/2/${data.image_id}/full/200,/0/default.jpg`,
    };
  });

  dispatch({
    type: GET_ART,
    payload: collection,
  });
};

const reducer = (state = intialState, action) => {
  switch (action.type) {
    case GET_ART:
      return action.payload;

    default:
      return state;
  }
};

export default reducer;
