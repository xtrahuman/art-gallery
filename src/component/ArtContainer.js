import React, { useEffect } from 'react';
import { Routes, Route } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import ArtList from './ArtList';
import DetailsPage from './DetailsPage';
import NotMatch from './NotMatch';
import { getArt } from '../redux/ArtList/Artlist';

const ArtContainer = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getArt());
  }, [dispatch]);

  return (
    <div className="allPages">
      <Routes>
        <Route path="/" element={<ArtList />} />
        <Route path="/details/:id" element={<DetailsPage />} />
        <Route path="*" element={<NotMatch />} />
      </Routes>
    </div>
  );
};

export default ArtContainer;
