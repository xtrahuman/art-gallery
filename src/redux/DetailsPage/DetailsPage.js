import { FetchArt } from '../ArtList/Artlist';

const EACH_ART = 'art-gallery/DetailsPage/EACH_ART';

const intialState = [];

export const eachArt = (id) => async (dispatch) => {
  const ArtDetail = await FetchArt();
  const collection = ArtDetail.data.map((data) => {
    const { title, id } = data;
    return {
      title,
      id,
      displayDate: data.date_display,
      artist: data.artist_display,
      country: data.place_of_origin,
      credit: data.credit_line,
      imgt: `https://www.artic.edu/iiif/2/${data.image_id}/full/100,/0/default.jpg`,
    };
  }).filter((art) => art.id === Number(id));

  dispatch({
    type: EACH_ART,
    payload: collection,
  });
};

const detailsReducer = (state = intialState, action) => {
  switch (action.type) {
    case EACH_ART:
      return action.payload;
    default:
      return state;
  }
};

export default detailsReducer;
