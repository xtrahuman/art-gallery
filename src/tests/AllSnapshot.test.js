import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configurStore';
import DetailsPage from '../component/DetailsPage';
import ArtList from '../component/ArtList';
import ArtContainer from '../component/ArtContainer';

describe('test all components snapshots', () => {
  test('matches details page', () => {
    const details = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <DetailsPage />
        </Router>
      </Provider>,
    ).toJSON();

    expect(details).toMatchSnapshot();
  });

  test('matches artlist', () => {
    const artlist = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <ArtList art={[
            {
              title: 'name',
              id: 124,
              displayDate: '1245',
            },
            {
              title: 'name',
              id: 124,
              displayDate: '1245',
            },
          ]}
          />
        </Router>
        ,
      </Provider>,
    ).toJSON();

    expect(artlist).toMatchSnapshot();
  });

  test('matches Artcontainer', () => {
    const Artcontain = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <ArtContainer />
        </Router>
      </Provider>,
    ).toJSON();

    expect(Artcontain).toMatchSnapshot();
  });
});
