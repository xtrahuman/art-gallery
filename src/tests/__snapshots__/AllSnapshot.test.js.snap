// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test all components snapshots matches Artcontainer 1`] = `
<div
  className="allPages"
>
  <div
    className="checkArt"
  >
    <div
      className="d-flex artPage"
    >
      <p>
        Art Gallery
      </p>
      <p>
        Art List
      </p>
      <div
        className="flex-setting d-flex"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
          >
            <path
              d="M0 0h24v24H0z"
            />
            <path
              d="M0 0h24v24H0z"
            />
            <path
              d="M0 0h24v24H0z"
            />
          </g>
          <path
            d="M12 14c1.66 0 3-1.34 3-3V5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3z"
          />
          <path
            d="M17 11c0 2.76-2.24 5-5 5s-5-2.24-5-5H5c0 3.53 2.61 6.43 6 6.92V21h2v-3.08c3.39-.49 6-3.39 6-6.92h-2z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58a.49.49 0 00.12-.61l-1.92-3.32a.488.488 0 00-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54a.484.484 0 00-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58a.49.49 0 00-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
          />
        </svg>
      </div>
    </div>
    <div
      className="artContainer"
    >
      <div
        className="ArtGallery"
      >
        <div
          className="listImg"
        >
          <img
            alt="art"
            src="gallery.svg"
          />
        </div>
        <div
          className="artText"
        >
          <h1>
            Art Gallery
          </h1>
          <p>
            0 
            ArtWork
          </p>
        </div>
      </div>
      <div
        className="searchInput d-flex"
      >
        <input
          onChange={[Function]}
          placeholder="search art"
        />
      </div>
      <div
        className="d-flex justify-row"
      >
        <div
          className="row"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`test all components snapshots matches artlist 1`] = `
Array [
  <div
    className="checkArt"
  >
    <div
      className="d-flex artPage"
    >
      <p>
        Art Gallery
      </p>
      <p>
        Art List
      </p>
      <div
        className="flex-setting d-flex"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
          >
            <path
              d="M0 0h24v24H0z"
            />
            <path
              d="M0 0h24v24H0z"
            />
            <path
              d="M0 0h24v24H0z"
            />
          </g>
          <path
            d="M12 14c1.66 0 3-1.34 3-3V5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3z"
          />
          <path
            d="M17 11c0 2.76-2.24 5-5 5s-5-2.24-5-5H5c0 3.53 2.61 6.43 6 6.92V21h2v-3.08c3.39-.49 6-3.39 6-6.92h-2z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58a.49.49 0 00.12-.61l-1.92-3.32a.488.488 0 00-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54a.484.484 0 00-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58a.49.49 0 00-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
          />
        </svg>
      </div>
    </div>
    <div
      className="artContainer"
    >
      <div
        className="ArtGallery"
      >
        <div
          className="listImg"
        >
          <img
            alt="art"
            src="gallery.svg"
          />
        </div>
        <div
          className="artText"
        >
          <h1>
            Art Gallery
          </h1>
          <p>
            0 
            ArtWork
          </p>
        </div>
      </div>
      <div
        className="searchInput d-flex"
      >
        <input
          onChange={[Function]}
          placeholder="search art"
        />
      </div>
      <div
        className="d-flex justify-row"
      >
        <div
          className="row"
        />
      </div>
    </div>
  </div>,
  ",",
]
`;

exports[`test all components snapshots matches details page 1`] = `
<div
  className="checkDetails"
>
  <div
    className="d-flex artPage"
  >
    <a
      aria-current="page"
      className="active"
      exact={true}
      href="/"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 256 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        />
      </svg>
    </a>
    <p>
      Art Details
    </p>
    <div
      className="flex-setting d-flex"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
        >
          <path
            d="M0 0h24v24H0z"
          />
          <path
            d="M0 0h24v24H0z"
          />
          <path
            d="M0 0h24v24H0z"
          />
        </g>
        <path
          d="M12 14c1.66 0 3-1.34 3-3V5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3z"
        />
        <path
          d="M17 11c0 2.76-2.24 5-5 5s-5-2.24-5-5H5c0 3.53 2.61 6.43 6 6.92V21h2v-3.08c3.39-.49 6-3.39 6-6.92h-2z"
        />
      </svg>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58a.49.49 0 00.12-.61l-1.92-3.32a.488.488 0 00-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54a.484.484 0 00-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58a.49.49 0 00-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
        />
      </svg>
    </div>
  </div>
</div>
`;
